tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -1969,6 +2049,51 @@ port::Status CudnnSupport::DoRnnBackwardImpl(
   return port::Status::OK();
 }
 
+port::Status CudnnSupport::DoCtcLossImpl(
+    Stream* stream, const CudnnRnnStateTensorDescriptor& probs_desc,
+    const DeviceMemoryBase probs_data,
+    const absl::Span<const int32>& labels_data,
+    const absl::Span<const int32>& labels_lengths_data,
+    const absl::Span<const int32>& input_lengths_data,
+    DeviceMemoryBase costs_data,
+    const CudnnRnnStateTensorDescriptor& grads_desc,
+    DeviceMemoryBase grads_data,
+    const CudnnCtcLossDescriptor& ctc_loss_desc,
+    ScratchAllocator* workspace_allocator) {
+  auto cudnn = cudnn_->GetHandle(parent_, stream);
+
+  SE_ASSIGN_OR_RETURN(DeviceMemory<uint8> workspace,
+                      CreateCtcLossWorkspace(stream, cudnn, ctc_loss_desc,
```

Comment:
Can you surface this to `ThenCtcLoss`? So that DnnSupport's DoCtcLoss takes a workspace memory, instead of a workspace allocator.